vc-worker/src/main/java/com/vmware/flowgate/vcworker/scheduler/job/VCDataService.java

```diff
@@ -168,11 +189,415 @@ public void checkAndUpdateIntegrationStatus(SDDCSoftwareConfig vc, String messag
+            needUpdate = needUpdate
+                  || (!licenseProductName.equals(oldHostInfo.getHypervisorLicenseProductName()));
+            hostInfo.setHypervisorLicenseProductName(licenseProductName);
+
+            String licenseProductVersion = aboutInfo.getLicenseProductVersion() == null ? ""
+                  : aboutInfo.getLicenseProductVersion();
+            needUpdate = needUpdate || (!licenseProductVersion
+                  .equals(oldHostInfo.getHypervisorLicenseProductVersion()));
+            hostInfo.setHypervisorLicenseProductVersion(licenseProductVersion);
+
+            String version = aboutInfo.getVersion() == null ? "" : aboutInfo.getVersion();
+            needUpdate = needUpdate || (!version.equals(oldHostInfo.getHypervisorVersion()));
+            hostInfo.setHypervisorVersion(version);
+         }
+
+         HardwareSummary hardwareSummary = summary.getHardware();
+         if (hardwareSummary != null) {
+
+            String hostModel = hardwareSummary.getModel() == null ? "" : hardwareSummary.getModel();
+            needUpdate = needUpdate || (!hostModel.equals(oldHostInfo.getHostModel()));
+            hostInfo.setHostModel(hostModel);
+
+            String vendor = hardwareSummary.getVendor() == null ? "" : hardwareSummary.getVendor();
+            needUpdate = needUpdate || (!vendor.equals(oldHostInfo.getHypervisorVendor()));
+            hostInfo.setHypervisorVendor(vendor);
+
+            needUpdate = needUpdate
+                  || (hardwareSummary.getNumCpuCores() != oldHostInfo.getCpuTotalCores());
+            hostInfo.setCpuTotalCores(hardwareSummary.getNumCpuCores());
+
+            needUpdate = needUpdate
+                  || (hardwareSummary.getNumCpuPkgs() != oldHostInfo.getCpuTotalPackages());
+            hostInfo.setCpuTotalPackages(hardwareSummary.getNumCpuPkgs());
+
+            needUpdate = needUpdate
+                  || (hardwareSummary.getNumCpuThreads() != oldHostInfo.getCpuTotalThreads());
+            hostInfo.setCpuTotalThreads(hardwareSummary.getNumCpuThreads());
+
+            needUpdate =
+                  needUpdate || (hardwareSummary.getCpuMhz() != oldHostInfo.getSingleCoreCpuMhz());
+            hostInfo.setSingleCoreCpuMhz(hardwareSummary.getCpuMhz());
+
+            needUpdate = needUpdate
+                  || (hardwareSummary.getMemorySize() != oldHostInfo.getMemoryCapacity());
+            hostInfo.setMemoryCapacity(hardwareSummary.getMemorySize());
+         }
+      }
+
+      NetworkInfo networkInfo = host.getConfig().getNetwork();
+      if (networkInfo != null) {
+         PhysicalNic[] physicalNics = networkInfo.getPnic();
+         int nicsNum = physicalNics.length;
+         if (nicsNum > 0) {
+            List<HostNic> hostNics = new ArrayList<HostNic>();
+
+            for (int i = 0; i < nicsNum; i++) {
+
+               HostNic hostNic = new HostNic();
+
+               hostNic.setMacAddress(physicalNics[i].getMac());
```

Comment:
Mac address is unique. 

1. compare map size

2. new {
  mac address ,check old map {
   
}else{
  oldmap put (new one)
}
}